apps/web/src/components/commands/tokens/Tokens.tsx

```diff
@@ -0,0 +1,67 @@
+import { DataPill } from "../../DataPill";
+import { useQuery } from "react-query";
+import { Tokens } from "./types";
+import { findToken } from "./helpers";
+import Image from "next/image";
+
+export type TokenWidgetProps = {
+  property: "price";
+  chainName: string;
+  token: string;
+  platforms: "native" | "basic";
+};
+
+export const TokenWidget = ({ property, ...props }: TokenWidgetProps) => {
+  switch (property) {
+    case "price":
+      return <TokenPriceWidget {...props} />;
+    default:
+      return null;
+  }
+};
+
+const TokenPriceWidget = ({
+  chainName,
+  token,
+  platforms,
+}: Pick<TokenWidgetProps, "chainName" | "token" | "platforms">) => {
+  const query = useQuery(
+    ["TOKEN", "PRICE", chainName, token, platforms],
+    async () => {
+      if (token.toLowerCase().trim() === "eth") {
+        platforms = "native";
+      }
+      const url = `https://api.portals.fi/v2/tokens?search=${token}&platforms=${platforms}&networks=${chainName}`;
+      const response = await fetch(url);
+      const json = await response.json();
```

Comment:
Type hint here maybe, `as PortalsResponse[]`? 

In the future we need response validation.

Reply:
Then you can type `findToken`